utils/py/model_catalog.py

```
@@ -0,0 +1,292 @@
+    """The committed YAML must be byte-identical to a fresh render of the vendored copy."""
+    try:
+        fresh = render_yaml(root)
+    except Exception as e:  # noqa: BLE001
+        return [f"render failed: {e}"]
+    with open(yaml_path(root), "rb") as f:
+        committed = f.read()
+    if fresh == committed:
+        return []
+    fresh_lines = fresh.decode("utf-8", "replace").splitlines()
+    committed_lines = committed.decode("utf-8", "replace").splitlines()
+    only_committed = [l for l in committed_lines if l not in fresh_lines]
```

Ruff (0.16.3)

[error] Ambiguous variable name: `l`

(E741)